### ISSUE TITLE
Allow inline latex, and make it easy to use

### ISSUE THREAD

**Issue by SylvainChevalier** [Metaculus/metaculus#971]
I couldn't find a way to insert inline latex math.

Adding math with the wysiwyg editor inserts a `<EmbeddedMathJax content="\[e=mc^2\]" />`. This is nonstandard and quite hostile.

Could we instead do [this](https://mdxjs.com/guides/math/)? (I suggest Katex rather than Mathjax since it's faster.)

**Comment by ncarazon** [comment]
Yeah, we did a solution with using the embedded JSX for math because mdx editor we use doesn't have a latex plugin and adding it would take more time to implement. But makes sense to do a proper implementation now.

**Comment by SylvainChevalier** [comment]
@ncarazon the version on play is great!

Some small tweaks:

- The formula edit box should grow with the content, or at least have a handle for me to resize it.
- It seems unnecessary to have two uis for editing the formulas: the modal when creating one and the inline text box when editing. I would recommend either:
  - Only have the modal.
  or
  - The modal has just two buttons: "inline", "block" ; pre-fill the formula text box with `p = q`.
- (_if we keep the modal_) The formulas in the modal are (needlessly) capitalised.
![image](https://github.com/user-attachments/assets/96024118-8e14-4221-90a7-1d24aaf20cfc)

**Comment by hickman-santini** [comment]
Ditto @SylvainChevalier re: edit box growing with the content, but then also in the modal, give me a scroll in cases like:
<img src="https://github.com/user-attachments/assets/37f184c4-fc9e-475b-901b-1714686e7dcb" width=250>

Just a little oddity, I don't think this is a problem? but lower-case greek letters like `\sigma` `\zeta` render as letters in the equation box:
<img src="https://github.com/user-attachments/assets/d92cc8e6-e657-4d95-a292-d2a2fc61a878" width=250>

I vote no editing in the modal, just "inline/block."

**Comment by SylvainChevalier** [comment]
Uppercase beta and uppercase zeta look like "B" and "Z" (resp.), so this is the capitalisation problem I listed above.
